### ISSUE TITLE
sdbus-c++-xml2cpp doesn't detect that input file doesn't exist

### ISSUE THREAD

**Issue by lejcik** [Kistler-Group/sdbus-cpp#83]
Hi Stano,

While setting up a new project I found that `sdbus-c++-xml2cpp` doesn't properly detect that input file doesn't exist. If I run this command (none of the files exist), then the output is confusing:

```
./sdbus-c++-xml2cpp --proxy=sample-client-glue.h /tmp/sample.xml
Parsing error: line 1, column 0: no element found
```

The problem is here: https://github.com/Kistler-Group/sdbus-cpp/blob/master/tools/xml2cpp-codegen/xml2cpp.cpp#L152

The condition should be `if (input.fail())`, as `fail()` covers both `failbit` and `badbit` `iostate`.  Then the output is correct:

```
./sdbus-c++-xml2cpp --proxy=sample-client-glue.h /tmp/sample.xml
Unable to open file /tmp/sample.xml
```

BR,
Vilo

**Comment by sangelovic** [comment]
Hey Vilo, thanks :) would you please submit a pull request with that change?

Thank you 🙄

**Comment by sangelovic** [comment]
Fixed by #84
